### DISCUSSION TITLE
The distribution within an element for a 1 D transport problem

### DISCUSSION THREAD

**Discussion by hhy2022** [idaholab/moose#22621] [Q&A General] [answered]
I am trying to solve for a 1 D transport problem. I got the result and found that there was a distribution inside a single element. I am a little confused about that.

The problem definition was like this: A species transported from the left to right. It was driven by the velocity. I had a left and right boundary condition given the amount of the species and would like to know the amount distribution along the domain. The geometry was 5 cm with 10 elements.

I could compile my application and get the result. The question was, there was a distribution inside a single element. But I thought it should be a unifrom distribution inside a single element and the changes should only located between elements since each element should be regared as a "unit" one.

I am trying to find out a explanation on this. MOOSE is computing the kernel at quadrature points. But I think this is not going to cause any difference inside the element. 

Any help are appreciated.

Yang

**Comment by GiudGiud** [comment, marked as answer]
Hello

What kind of variables are you using?
The default variable type is 1st order lagrange, which is linear across each element

Guillaume

**Reply by hhy2022** [reply]
I didn't specify the variable type. When I defined the variable, I just gave the name. 
So my variable is 1st order lagrange, right? Does this mean the shape functions are first order? 
Beside, how the result for one single element was comupted? Is there any functions or kernels that I can use to compute the relationship between the species ammout with the x axis?

**Reply by GiudGiud** [reply]
yes then it is the default, and the shape functions are first order. 
The solution is defined continuously over the domain. If you want the average over a single element you ll have to project it to a CONSTANT MONOMIAL using a SelfAux

**Reply by hhy2022** [reply]
OK. Thank you very much for your help.

Besides, is there any syntax specified the number of quadrature points? What is the relationship between the qp point result and the result for one element?

**Reply by GiudGiud** [reply]
There is the quadrature block under the executioner. 
https://mooseframework.inl.gov/source/actions/SetupQuadratureAction.html

The qp value can be computed from the nodal values and vice-versa. We output and store nodal values, computations like the residuals use the qp values

**Reply by hhy2022** [reply]
Then what is the relationship between nodal and element? For a 1 D problem, if I had 10 elements, does it mean I had 11 nodes? If qp can be computed by nodal values, then there would be 11 qp values?

**Reply by GiudGiud** [reply]
10 elements -> 11 nodes but for the Qps it will depend on the quadrature order

**Reply by hhy2022** [reply]
I see. So the Qp only relevant to the quadrature order, right? I mean, if I defined the order of my variable, then the qps for each element will be decided. Is it correct? Or is there another function to specify the qp.

**Reply by GiudGiud** [reply]
no the variable order and the quadrature order are different.
the quadrature may be changed in Executioner/quadrature.

It is also adjusted by default to be able to integrate the variable family types used

**Reply by hhy2022** [reply]
Usually we don't need to define the qp right? Or it is defined by the variable family? Are they relevant to the geometry/mesh ?

Sorry I am kind of confused about this part. Especially when I got a result plot in peacock with colorbar. I am curious on how these values are computed.

**Reply by GiudGiud** [reply]
The Qps, and their weight, are defined by the quadrature.

What you are seeing in paraview are the nodal value, and paraview knows to draw them continuously for point data

**Reply by hhy2022** [reply]
In this case, is there any methods to draw a line figure? For example, if I am solving for a transient problem, what I can draw right now is the plot for the whole domain with the changes of time. Can I know the changes in the x direction with specific time steps? Do I need to use some postprocesses or paraview has the function to deal with the point data?

**Reply by GiudGiud** [reply]
Paraview can draw point data over time
CSV output using a PointValueSampler or LineValueSampler is also an option
